sys/linux/rdma.txt

```diff
@@ -0,0 +1,1464 @@
+	count	ib_uverbs_flow_spec_action_count
+]
+
+ib_uverbs_flow_attr {
+	type		int32[0:3]
+	size		bytesize[flow_specs, int16]
+	priority	int16[0:7]
+	num_of_specs	len[flow_specs, int8]
+	reserved1	const[0x0, int8]
+	reserved2	const[0x0, int8]
+	port		int8
+	flags		flags[ib_flow_flags, int32]
+	flow_specs	array[union_ib_flow_spec]
+}
+
+create_flow_cmd_ex {
+	comp_mask	const[0x0, int32]
+	qp_handle	qp_handle
+	flow_attr	ib_uverbs_flow_attr
+}
+
+create_flow_resp {
+	comp_mask	int32
+	flow_handle	flow_handle
+}
+
+create_flow_cmd {
+	command			const[IB_USER_VERBS_EX_CMD_CREATE_FLOW, int32]
+	in_words		bytesize8[flow_ex, int16]
+	out_words		const[0x1, int16]
+	response		ptr[out, create_flow_resp]
+	provider_in_words	const[0x0, int16]
+	provider_out_words	const[0x0, int16]
+	reserved		const[0x0, int32]
+	flow_ex			create_flow_cmd_ex
+}
+
+destroy_flow_cmd {
+	command			const[IB_USER_VERBS_EX_CMD_DESTROY_FLOW, int32]
+	in_words		const[0x6, int16]
+	out_words		const[0x0, int16]
+	response		const[0x0, int64]
+	provider_in_words	const[0x0, int16]
+	provider_out_words	const[0x0, int16]
+	reserved		const[0x0, int32]
+	comp_mask		const[0x0, int32]
+	flow_handle		flow_handle
+}
+
+# ioctl interface
+
+ib_uverbs_attr_flags = UVERBS_ATTR_F_MANDATORY
+ib_read_counters_flags = IB_UVERBS_READ_COUNTERS_PREFER_CACHED
+resource vcontext_handle[int32]
+
+ib_uverbs_create_counters_cmd {
+	length		bytesize8[ib_uverbs_create_counters_cmd, int16]
+	object_id	const[UVERBS_OBJECT_COUNTERS, int16]
+	method_id	const[UVERBS_METHOD_COUNTERS_CREATE, int16]
+	num_attrs	const[1, int16]
+	reserved0	const[0x0, int64]
+	driver_id	const[RDMA_DRIVER_MLX5, int32]
+	reserved1	const[0x0, int32]
+
+	attr_id		const[UVERBS_ATTR_CREATE_COUNTERS_HANDLE, int16]
+	len		int16
+	flags		flags[ib_uverbs_attr_flags, int16]
+	elem_id		int8
+	reserved2	const[0x0, int8]
+	reserved3	const[0x0, int16]
+	vcontext_handle	vcontext_handle[opt]
+	more_data	int32
+}
+
+ib_uverbs_destroy_counters_cmd {
+	length		bytesize8[ib_uverbs_destroy_counters_cmd, int16]
+	object_id	const[UVERBS_OBJECT_COUNTERS, int16]
+	method_id	const[UVERBS_METHOD_COUNTERS_DESTROY, int16]
+	num_attrs	const[1, int16]
+	reserved0	const[0x0, int64]
+	driver_id	const[RDMA_DRIVER_MLX5, int32]
+	reserved1	const[0x0, int32]
+
+	attr_id		const[UVERBS_ATTR_DESTROY_COUNTERS_HANDLE, int16]
+	len		int16
+	flags		flags[ib_uverbs_attr_flags, int16]
+	elem_id		int8
+	reserved2	const[0x0, int8]
+	reserved3	const[0x0, int16]
+	vcontext_handle	vcontext_handle
+	more_data	int32
+}
+
+# in UVERBS_ATTR_READ_COUNTERS_FLAGS, assuming that sizeof(uint32_t is 4)
+ib_uverbs_read_counters_cmd {
+	length		bytesize8[ib_uverbs_read_counters_cmd, int16]
+	object_id	const[UVERBS_OBJECT_COUNTERS, int16]
+	method_id	const[UVERBS_METHOD_COUNTERS_READ, int16]
+	num_attrs	const[0x3, int16]
+	reserved0	const[0x0, int64]
+	driver_id	const[RDMA_DRIVER_MLX5, int32]
+	reserved1	const[0x0, int32]
+
+	attr_id0	const[UVERBS_ATTR_READ_COUNTERS_HANDLE, int16]
```

Comment:
We could make this an actual array:
```
attrs array[....]
```
and then use `len[attars]` for `num_attrs`.
This would allow fuzzer to generate more interesting options.